### PR TITLE
Add null check for images in loading skeleton

### LINKED ISSUE #2923: `TypeError` in VImageGrid
## Description

> Cannot read properties of null (reading 'length')

It seems that the `images` array can be `null` at some point, causing this error. This is the only variable using the length property in this file. We should check the variable is not null before trying to access length.

https://github.com/WordPress/openverse/blob/90dcb5224704dcfe5d5fce192ba4702f90bc5d7e/frontend/src/components/VSearchResultsGrid/VImageGrid.vue#L3-L6

## Additional context

See the [Sentry](https://openverse.sentry.io/share/issue/659a5d4eb4344dceb7ba23de1fb06a22/) details.

### PR DESCRIPTION
**Pull request by akhilsrivatsa**
## Fixes

Fixes #2923 by @krysal

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by akhilsrivatsa** [comment]
Kinda new to this whole open source thing and its my first ever PR. Apologies if I screwed something up! @obulat @dhruvkb

**Comment by obulat** [comment]
Hi @ akhilsrivatsa! Welcome to Openverse 🎉 

Which issue were you trying to address with this PR? I think the `null` check for images was already added, but if there is an open issue for that, could you please share it?
When you fill in the PR description, the `Fixes` section is really helpful because it links to the issue the PR solves. You can add the number of the issue and the issue author instead of the placeholder in brackets, and GitHub automatically adds the links. This is how it would look in the PR edit view:
```
Fixes #11111 by @obulat
```
If there is no issue, you can also find another one you can tackle by looking through the [issues tagged with `good first issue` label](https://github.com/WordPress/openverse/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22). Please, feel free to ask for help whenever you need it.

**Comment by akhilsrivatsa** [comment]
@obulat Here is the issue - #2923

File: frontend/src/components/VSearchResultsGrid/VImageGrid.vue
Comment (obulat) [inline frontend/src/components/VSearchResultsGrid/VImageGrid.vue:4]:
```suggestion
      v-if="!images?.length && !fetchState.isFinished"
```
Optional chaining operator `?`[^1] can make this expression cleaner. It returns `undefined` if `images` is null or undefined, and does not try to access the `.length` property.

[^1]: https://javascript.info/optional-chaining

**Review by obulat — CHANGES_REQUESTED** [review]
Thank you for linking to the issue!
I added a suggestion to make the check cleaner using the `?` operator (optional chaining), and approved the PR.
Note that you need 2 PR approvals to get it merged.

**Review by obulat — APPROVED** [review]

**Comment by krysal** [comment]
@obulat A test is failing after the modification on the strategy, I certainly find it strange for such a simple change! Could you take a second look on this?

**Comment by akhilsrivatsa** [comment]
@obulat I just my checked github and I see that you've fixed it. Sorry for the delay.

**Comment by obulat** [comment]
@akhilsrivatsa, it appears that the changes on main branch somehow affected the CI in your PR, and that caused a failure. I'm sorry for this confusion. I reverted my change that added an optional chaining, and now the PR is ✅  Thank you for your contribution!
@krysal, the PR is now ready for the second review. I think the testing failure was due to the changes on main, not changes in this PR.

File: frontend/src/components/VSearchResultsGrid/VImageGrid.vue
Comment (dhruvkb) [inline frontend/src/components/VSearchResultsGrid/VImageGrid.vue:4]: This also accounts for undefined.
```suggestion
      v-if="images && !images.length && !fetchState.isFinished"
```

**Review by dhruvkb — APPROVED** [review]
LGTM, except for a small nit.

**Pull request merged**
